src/compactor_state.rs

```diff
@@ -145,11 +150,12 @@ impl CompactorState {
         Ok(id)
     }
 
-    pub(crate) fn merge_db_state(&mut self, updated_state: &CoreDbState) {
+    pub(crate) fn merge_remote_manifest(&mut self, mut remote_manifest: DirtyManifest) {
```

Comment:
I wonder if this and the similar function in `CoreDbState` can be combined into a single `DirtyManifest.merge` method? That would remove the context-dependent assumptions about which parts of the state were udpated.

Reply:
Yeah, but I'd like to tackle that in a later PR. The way I think we should structure it is to define some type that encapsulates a `DirtyManifest` along with a log of mutations that are made to it. To merge we would reapply the log of mutations. We'd also need some mechanism to clear the log when the dirty manifest is written out to the object store.

Reply:
Sounds good.